lib/Component.lua

```diff
@@ -424,4 +431,35 @@ function Component:_unmount()
 	self._handle = nil
 end
 
+--[[
+	Performs property validation, if it is enabled.
+]]
+function Component:_validateProps(props)
+	if not GlobalConfig.getValue("propValidation") then
+		return
+	end
+
+	local validator = self.validateProps
+
+	if validator == nil then return end
+	if typeof(validator) ~= "function" then
+		-- Hide as much as possible about error location, since this message
+		-- occurs from several possible call sites with different stack traces.
+		error(("The value of validateProps must be a function, but it is a %s.\n"..
+			"Check the definition of the component extended at:\n%s"):format(
+			typeof(validator),
+			self._extendTraceback),
+		0)
+	end
```

Comment:
Is this totally necessary?  We don't bother doing this with lifecycle methods or things like `defaultProps` (making sure it's a table).

I can see the argument for doing it here since you're already opting into validation, just wondering what the thought process is since it's not something we do elsewhere.

Reply:
Most of the time when `validateProps` is specified, it's not being written as a method:

```lua
function SomeComponent.validateProps(props)
    -- do some validation work
end
```

Instead, it's usually composed from a bunch of smaller validators, and the result of the composition is assigned. When you're using PropTypes it looks something like this:

```lua
SomeComponent.validateProps = PropTypes.object {
    someKey = PropTypes.number,
    someOtherKey = PropTypes.string,
}
```

The type check is an extra guard against `validateProps` being assigned a non-function, since it's probably not going to be declared as a method of the component class, but as a field of it.

Reply:
Gotcha, that makes sense!